crm_claim_rma/wizards/claim_make_picking.py

```diff
@@ -0,0 +1,235 @@
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import time
+
+from openerp import models, fields, exceptions, api, workflow, _
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT as DT_FORMAT
+
+
+class ClaimMakePicking(models.TransientModel):
+    _name = 'claim_make_picking.wizard'
+    _description = 'Wizard to create pickings from claim lines'
+
+    def _get_common_field_from_lines(self, lines, field):
+        """
+        Read the supplied field from all the lines.
+        Lines that do not have the supplied field set are ignored.
+        If all the remaining lines have the same value for the field,
+        that value is returned, else False is returned
+        """
+        field_values = lines.mapped(field)
+        return field_values[0] if len(field_values) == 1 else False
+
+    def _get_common_dest_location_from_line(self, lines):
+        return self._get_common_field_from_lines(
+            lines, 'location_dest_id.id')
+
+    def _get_common_partner_from_line(self, lines):
+        return self._get_common_field_from_lines(
+            lines, 'warranty_return_partner.id')
+
+    def _default_claim_line_source_location_id(self):
+        picking_type = self.env.context.get('picking_type')
+        partner_id = self.env.context.get('partner_id')
+        warehouse_id = self.env.context.get('warehouse_id')
+
+        if picking_type == 'out' and warehouse_id:
+            return self.env['stock.warehouse'].browse(warehouse_id).lot_stock_id
+
+        if partner_id:
+            partner = self.env['res.partner'].browse(partner_id)
+            return partner.property_stock_customer
+
+        return False
+
+    def _default_claim_line_dest_location_id(self):
+        """Return the location_id to use as destination.
+
+        If it's an outoing shipment: take the customer stock property
+        If it's an incoming shipment take the location_dest_id common to all
+        lines, or if different, return None.
+        """
+
+        picking_type = self.env.context.get('picking_type')
+        partner_id = self.env.context.get('partner_id')
+
+        if picking_type == 'out' and partner_id:
+            return self.env['res.partner'].browse(
+                partner_id).property_stock_customer
+
+        if picking_type == 'in' and partner_id:
+            # Add the case of return to supplier !
+            lines = self._default_claim_line_ids()
+            return self._get_common_dest_location_from_line(lines)
```

Comment:
Does this mean you sometimes return an id and sometimes a browse object? Maybe worth making the `_get_common_...` functions return a browse object?